### PR TITLE
add DIC/config support

### PR DESCRIPTION
**Pull request by sandrokeil**
~~Don't merge, there are some problems. The test services are not available in the bundle `load()` method. See comments below. I want to aviod an environment check in the bundle class.~~

[x] XML Config for event store
[x] YML Config for event store

### CODE REVIEW

File: test/DependencyInjection/AbtractServiceBusExtensionTestCase.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    private function loadContainer($fixture, CompilerPassInterface $compilerPass = null)
+    {
+        $container = $this->getContainer();
+
+        $container->registerExtension(new ProophEventStoreExtension());
+
+        $loadXml = new XmlFileLoader($container, new FileLocator(dirname(__DIR__).'/Resources/config'));
+        $loadXml->load('services.xml');
+
```
Comment (sandrokeil) [inline test/DependencyInjection/AbtractServiceBusExtensionTestCase.php]: Test services are not available in extension. How can I inject services for testing only?

File: test/Resources/config/services.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Metadata\BlackHole"
+                 class="ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Metadata\BlackHole"/>
+        <service id="ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Plugin\BlackHole"
+                 class="ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Plugin\BlackHole"/>
+        <service id="ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Adapter\BlackHole"
+                 class="ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Adapter\BlackHole"/>
+    </services>
```
Comment (sandrokeil) [inline test/Resources/config/services.xml]: I need this service configuration for testing only. Any ideas how to inject them, so they are available in the bundle?

File: src/DependencyInjection/ProophEventStoreExtension.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        $def = $container->getDefinition('prooph_event_store.' . $type);
+        $def->setClass($class);
+
+        foreach ($config[$typePlural] as $name => $options) {
+            $this->loadEventStore($type, $name, $options, $container);
+        }
+    }
+
+    /**
+     * Initializes specific event store class with plugins and routes. Each class dependency must be set via a container
```
Comment (codeliner) [inline src/DependencyInjection/ProophEventStoreExtension.php]: remove "and routes"

**Comment by sandrokeil** [comment]
@codeliner It's done for the first alpha release :tada:

File: src/DependencyInjection/ProophEventStoreExtension.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+use Prooph\EventStore\EventStore;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * Defines and load message bus instances.
```
Comment (codeliner) [inline src/DependencyInjection/ProophEventStoreExtension.php]: s/message bus/event store

**Pull request merged**